blaze-server/src/main/scala/org/http4s/server/blaze/Http1ServerStage.scala

```diff
@@ -311,10 +326,8 @@ private[blaze] class Http1ServerStage[F[_]](
 
   private def cancel(): Unit =
     cancelToken.foreach { token =>
-      F.runAsync(token) {
-        case Right(_) => IO(logger.debug("Canceled request"))
-        case Left(t) => IO(logger.error(t)("Error canceling request"))
-      }.unsafeRunSync()
+      Await.result(token(), Duration.Inf)
```

Comment:
Is `Duration.Inf` safe here?

Reply:
`fa.unsafeRunSync` is ultimately an `Await.result(fa, Duration.Inf)` in both CE2 and CE3, so it does preserve semantics

Reply:
Okay, that's a good point.